### PR TITLE
VZ 5193: Resizable Data Nodes

### PR DESCRIPTION
**Pull request by anders-swanson**
Picks up most recent VMO image to allow OpenSearch data nodes to be resized after installation. 
CRD: Takes the latest CRD from VMO that supports per-node storage options.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/verrazzano/vmi.go [hunk 1]
```diff
@@ -172,6 +180,37 @@ func newOpenSearch(cr *vzapi.Verrazzano, storage *resourceRequestValues, vmi *vm
...
+				Size: storage.Storage,
+			}
+		}
+		return nodeStorage
+	}
+	opensearch.MasterNode.Storage = setVolumeClaimOverride(opensearch.MasterNode.Storage, hasMasterOverride)
+	opensearch.DataNode.Storage = setVolumeClaimOverride(opensearch.DataNode.Storage, hasDataOverride)
+
+	if vmi != nil {
+		// We currently do not support resizing master node PVC
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/verrazzano/vmi.go:201]: we should enforce this in the validators...we can look into that in my PR or the next pass.

File: platform-operator/controllers/verrazzano/component/verrazzano/vmi_test.go [hunk 2]
```diff
@@ -121,10 +122,36 @@ func TestOpenSearchInvalidArgs(t *testing.T) {
 		},
 	}
 
-	_, err := newOpenSearch(crBadArgs, r, nil, false)
+	_, err := newOpenSearch(crBadArgs, r, nil, false, false)
 	assert.Error(t, err)
 }
 
+func TestNewOpenSearchWithExistingVMI(t *testing.T) {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/verrazzano/vmi_test.go:129]: Nit, should have a test header

**Review by michael-cico — APPROVED** [review]

**Review by mgianatagh — APPROVED** [review]

File: platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml [hunk 3]
```diff
@@ -16,452 +16,452 @@ spec:
...
-                            type: string
-                        type: object
-                      storage:
-                        description: Storage details
-                        properties:
-                          availabilityDomain:
-                            type: string
-                          pvcNames:
-                            items:
+              type: string
```
Comment (kminder) [inline platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:37]: It is hard to tell what the actual CRD changes are.  Can you summarize in the PR description?

Comment (anders-swanson) [reply platform-operator/helm_config/charts/verrazzano/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:37]: Updated. It's the latest from VMO with Per-Node storage options.
